converters/pokemon.py

```diff
@@ -288,20 +298,56 @@ def convert_pdata(input_csv, header=DEFAULT_HEADER):
         evolve = Evolve(header, pokemon_list)
         filter_data = FilterData(header)
         index_order = IndexOrder(header)
+        
+        poke_by_name = {}
 
+        # Collect all the rows into Pokemon types
         for index, row in enumerate(reader, 1):
             if not row:
                 continue
 
             # Each row is one Pokemon
             poke = Pokemon(header)
             poke.setup(row)
+            poke_by_name[poke.name] = poke
+
+        # Some rows are variants of a single pokemon type. Let's go collect those
+        variant_by_type = {}
+        def_poke_by_type = {}
+        def_type_by_variant = {}
+        # TODO: In the future we may have other variants, like Alolan forms or something.
+        # It's unclear what those might look like from a data perspective
+        for name,variants in util.VARIANT_DATA.items():
+            if type(variants) is list:
+                for d in variants:
+                    if d["name"] in poke_by_name:
+                        poke = poke_by_name[d["name"]]
+                        if "default" in d and d["default"]:
+                            poke.name = name
+                            poke.add_default_variant(d["variant_name"])
+                            def_type_by_variant[name] = d["name"]
+                        else:
+                            # Store for later, after the default variant has been collected
+                            variant_by_type[d["name"]] = d
+                            def_poke_by_type[d["name"]] = name
+                    else:
+                        raise Exception("When searching for variants, could not find pokemon of type '" + d["name"] + "'")
```

Comment:
Python have quite a few ways format strings, this is my favorite as I find it cleaner and easy to read `raise Exception(f"When searching for variants, could not find pokemon of type {data['name']}")`